RealmSwift-swift2.0/RealmCollectionType.swift

```diff
@@ -41,6 +41,120 @@ public final class RLMGenerator<T: Object>: GeneratorType {
+    /// are in the collection and/or modified one or more of the objects in the
+    /// collection.
+    ///
+    /// All three of the change arrays are always sorted in ascending order.
+    ///
+    /// @param deletions The indices in the previous version of the collection
+    ///                  which were removed from this one.
+    /// @param insertion The indices in the new collection which were added in
+    ///                  this version.
+    /// @param insertion The indices of the objects in the new collection which
+    ///                  were modified in this version.
+    case Update(T, deletions: [Int], insertions: [Int], modifications: [Int])
+
+    /// If an error occurs, notification blocks are called one time with a
+    /// .Error result and an NSError with details. Currently the only thing
+    /// that can fail is opening the Realm on a background worker thread to
+    /// calculate the change set.
+    case Error(NSError)
+
+    static func fromObjc(value: T, change: RLMCollectionChange?, error: NSError?) -> RealmCollectionChange {
+        if let error = error {
+            return .Error(error)
+        }
+        if let change = change {
+            return .Update(value,
+                deletions: change.deletions as! [Int],
+                insertions: change.insertions as! [Int],
+                modifications: change.modifications as! [Int])
+        }
+        return .Initial(value)
+    }
+}
+
+/**
+ RealmCollectionChange is passed to the notification blocks for Realm
+ collections, and reports the current state of the collection and what changes
+ were made to the collection since the last time the notification was called.
+
+ This differs from RealmCollectionChange<T> only in that it reports changes as
+ arrays of index paths suitable for use with UITableView rather than arrays of
+ Ints. A complete example of updating a UITableView given a change object is:
+
+        self.notificationToken = results.addNotificationBlock { (change: RealmCollectionChangePaths) in
+            switch change {
+            case .Initial:
+                // Results are now populated and can be accessed without blocking the UI
+                self.tableView.reloadData()
+                break
+            case .Update(_, let deletions, let insertions, let modifications):
+                // Query results have changed, so apply them to the TableView
+                self.tableView.beginUpdates()
+                self.tableView.insertRowsAtIndexPaths(insertions, withRowAnimation: .Automatic)
+                self.tableView.deleteRowsAtIndexPaths(deletions, withRowAnimation: .Automatic)
+                self.tableView.reloadRowsAtIndexPaths(modifications, withRowAnimation: .Automatic)
+                self.tableView.endUpdates()
+                break
+            case .Error(let err):
+                // An error occured while opening the Realm file on the background worker thread
```

Comment:
I noticed this when copying this code for the documentation: 'occurred' needs another 'r' there.